iroh-metrics/src/bitswap.rs

```diff
@@ -65,6 +68,60 @@ impl Metrics {
     }
 }
 
+impl MetricsRecorder for Metrics {
```

Comment:
what if the api was instead

```rust
BitswapMetrics::Requests.record(value);
```

this way you could avoid the `name` matching

Reply:
alternatively, we could make the api more like this


```rust
// this is similar to how log! works

record!(Bitswap::Requests, 15);
record!(Store::GetRequests, 4.3); // with macros we can use different types here


// could have shortcuts generated if we want to
bitswap_requests!(15);
store_get_requests!(4.3);
```

Reply:
K, think I got it in a solid place now and am happy with it.